core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -1537,4 +1543,13 @@ public static boolean isFaultTolerantExecutionEventDriverSchedulerEnabled(Sessio
     {
         return session.getSystemProperty(FAULT_TOLERANT_EXECUTION_EVENT_DRIVEN_SCHEDULER_ENABLED, Boolean.class);
     }
+
+    public static boolean isFaultTolerantExecutionForcePreferredWritePartitioningEnabled(Session session)
+    {
+        if (!isFaultTolerantExecutionEventDriverSchedulerEnabled(session)) {
+            // supported only in event driven scheduler
+            return false;
```

Comment:
should we throw instead?

Reply:
Hopefully we will remove the legacy scheduler soon. I added this more as of a precaucion in case somebody needs to revert to the old scheduler what could result into skew issues.